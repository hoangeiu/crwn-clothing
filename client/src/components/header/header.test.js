import React from "react";
import { shallow } from "enzyme";

import { Header } from "./header.component";
import CartDropdown from "../cart-dropdown/cart-dropdown.component";

const findTestByAttr = (wrapper, val) => wrapper.find(`[data-test="${val}"]`);

describe("Header component", () => {
  let wrapper;
  let mockSignOutStart;
  beforeEach(() => {
    mockSignOutStart = jest.fn();

    const mockProps = {
      currentUser: {
        uid: "123",
      },
      hidden: true,
      signOutStart: mockSignOutStart,
    };

    wrapper = shallow(<Header {...mockProps} />);
  });

  describe("Header component", () => {
    it("should Header component render without error", () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe("check currentUser prop", () => {
      it("render SIGNOUT text if currentUser exist", () => {
        const signOutHeader = findTestByAttr(wrapper, "signOut");

        expect(signOutHeader.text()).toBe("SIGN OUT");
        // expect(wrapper.find("OptionLink").at(2).text()).toBe("SIGN OUT");
      });

      it("should call signOutStart if button SIGN OUT is clicked", () => {
        const signOutHeader = findTestByAttr(wrapper, "signOut");
        signOutHeader.simulate("click");

        expect(mockSignOutStart).toHaveBeenCalled();
      });

      it("render SIGNIN text if currentUser not exist", () => {
        const mockProps = {
          currentUser: null,
          hidden: true,
          signOutStart: mockSignOutStart,
        };

        wrapper = shallow(<Header {...mockProps} />);

        const signInHeader = findTestByAttr(wrapper, "signIn");

        expect(signInHeader.text()).toBe("SIGN IN");
      });
    });

    describe("check hidden prop", () => {
      it("should not render CartDropdown if hidden is true", () => {
        expect(wrapper.exists(CartDropdown)).toBeFalsy();
      });

      it("should render CartDropdown if hidden prop is false ", () => {
        const mockProps = {
          currentUser: {
            uid: "123",
          },
          hidden: false,
          signOutStart: mockSignOutStart,
        };

        wrapper = shallow(<Header {...mockProps} />);

        expect(wrapper.find(CartDropdown).length).toBe(1);
      });
    });
  });
});
