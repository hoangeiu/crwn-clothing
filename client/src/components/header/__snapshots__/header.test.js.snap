// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header component should Header component render without error 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    <ForwardRef(SvgCrown)
      className="logo"
    />
  </Styled(Link)>
  <styled.div>
    <Styled(Link)
      to="/shop"
    >
      SHOP
    </Styled(Link)>
    <Styled(Link)
      to="/shop"
    >
      CONTACT
    </Styled(Link)>
    <Styled(Link)
      as="div"
      data-test="signOut"
      onClick={[MockFunction]}
    >
      SIGN OUT
    </Styled(Link)>
    <Connect(CartIcon) />
  </styled.div>
</styled.div>
`;
